cirq-core/cirq/circuits/circuit_operation.py

```diff
@@ -130,21 +135,30 @@ def __post_init__(self):
             raise TypeError(f'Expected circuit of type FrozenCircuit, got: {type(self.circuit)!r}')
 
         # Ensure that the circuit is invertible if the repetitions are negative.
-        if self.repetitions < 0:
-            try:
-                protocols.inverse(self.circuit.unfreeze())
-            except TypeError:
-                raise ValueError('repetitions are negative but the circuit is not invertible')
-
-        # Initialize repetition_ids to default, if unspecified. Else, validate their length.
-        loop_size = abs(self.repetitions)
-        if not self.repetition_ids:
-            object.__setattr__(self, 'repetition_ids', self._default_repetition_ids())
-        elif len(self.repetition_ids) != loop_size:
-            raise ValueError(
-                f'Expected repetition_ids to be a list of length {loop_size}, '
-                f'got: {self.repetition_ids}'
-            )
+        if isinstance(self.repetitions, float):
+            if abs(self.repetitions - round(self.repetitions)) < 0.001:
+                object.__setattr__(self, 'repetitions', round(self.repetitions))
+        if isinstance(self.repetitions, (int, np.integer)):
```

Comment:
Might be nice to store this tuple of types in a module variable, since it's used in a bunch of places:
```python
INT_CLASSES = (int, np.integer)
INT_TYPE = Union[int, np.integer]
```
I must say it's rather unfortunate that we can't use a `Union` like that directly in an `isinstance` check.